Display image preceded by exclamation #387

sup95 opened the pull request:
I have made the necessary changes in the code to resolve the bug. The image link is now displayed even when preceded by an exclamation. Do let me know if there are any more changes I should be making, thank you.

smarx commented:
*Automated message from Dropbox CLA bot*

@sup95, it looks like you've already signed the Dropbox CLA. Thanks!

timabbott commented:
@sup95 It looks like you accidentally have the code for #1400 in this PR; can you rebase your code and update this PR with the code that you want to run?

sup95 commented:
@timabbott I am so sorry, I get confused with git sometimes. I have rebased the code and updated the PR. I still somehow don't feel very confident if I have done the right thing. Do let me know if I have messed up something and modifications need to be made. Thank you.

timabbott commented:
Yeah, it looks like you've ended up with a bunch of the changes we've merged into master recently incorporated into your branch.  I suspect if you do a `git rebase upstream/master` they will be automatically dropped and you'll be left with just your own changes.  But I'd recommend you get a graphical git client and read https://github.com/zulip/zulip/pull/1879/files to learn how to fix this situation, since it'll be much easier if you use a graphical git viewer like `gitk` to understand what's happening as you work on fixing it.

brainwane commented:
@sup95 If you need more help on getting your pull request properly sorted out, check out https://zulip.tabbott.net/#narrow/stream/development.20help and [the draft git workflow documentation](https://zulip-ck.readthedocs.io/en/1754-docs-add-git-workflow/git-guide.html) that @christi3k is working on.

sup95 commented:
Sorry for the late response. I realised my master was really messed up. I did rebase the code, but something kept going wrong. I have now made the change in a separate branch and made another PR. Do have a look at it. :) So I'll close this PR I suppose, I am sorry for creating an extra PR, I shall be extra careful from now on with rebasing. Thank you.

timabbott commented:
OK, not a big deal.  For next time, there's a better solution to this problem.  Using e.g. `git push origin +localbranch:remotebranch` you can update a remote branch (and thus a PR) to be based off a different branch in your local repository; so you'd do all of the things you did this time, but instead of closing the PR and creating a new one, you'd just do a force push with `git`.
